crates/solver/src/driver.rs

```diff
@@ -630,6 +634,74 @@ fn orders_with_price_estimates(
     orders
 }
 
+// In case we cannot estimate the price for a token but there is only one pair traded in the batch the accuracy of the price estimate for ranking a solution doesn't matter that much (surplus is not competing with other tokens).
+// Therefore we can deduct a price estimate by looking at the average limit price of the orders on the single token pair
+fn augment_prices(orders: &[LimitOrder], prices: &mut HashMap<H160, BigRational>) {
+    if orders.is_empty() {
+        return;
+    }
+    let first_order = orders.first().unwrap();
```

Comment:
ubernit: A slightly more Rust-y way to do this IMO in order to avoid `unwrap` is:
```suggestion
    let first_order = match orders.first() {
        Some(order) => order,
        None => return,
    };
```

Reply:
Or we can accept a reference to a single order for this function, and do `if let` at the callsite.

Reply:
Oh no we can't, never mind.